src/sql/platform/telemetry/telemetry.contribution.ts

```diff
@@ -13,42 +13,124 @@ import { IStorageService, StorageScope } from 'vs/platform/storage/common/storag
 			telemetryService.publicLog('telemetry.monthlyUse', { monthlyFirstUse: true });
-			storageService.store('telemetry.monthlyLastUseDate', today, StorageScope.GLOBAL);
-		}
+			// the month changed, so send the user usage type event based on monthly count for last month
+			// and reset the count for this month
+			let lastMonthDate = new Date(monthlyLastUseDate);
+			this.sendUsageEvent(monthlyUseCount, lastMonthDate);
+
+			const wasActiveLastMonth: boolean = storageService.getBoolean('telemetry.wasActiveLastMonth', StorageScope.GLOBAL, false);
+			const isActiveThisMonth: boolean = storageService.getNumber('telemetry.monthlyUseCount', StorageScope.GLOBAL, 0) >= 1;
+			const isContinuing: boolean = wasActiveLastMonth && isActiveThisMonth;
+			const isReturning: boolean = !wasActiveLastMonth && isActiveThisMonth;
+
+			if (firstTimeUser) {
+				// new user
+				telemetryService.publicLog('telemetry.userGrowthType', {
+						userGrowthType: UserGrowthType.NewUser, month: lastMonthDate.getMonth().toString(), year: lastMonthDate.getFullYear().toString()});
+			}
 
+			if (isContinuing) {
+				// continuing user
+				telemetryService.publicLog('telemetry.userGrowthType',
+					{ userGrowthType: UserGrowthType.ContinuingUser, month: lastMonthDate.getMonth().toString(), year: lastMonthDate.getFullYear().toString()});
+			}
+
+			if (isReturning) {
+				// returning user
+				telemetryService.publicLog('telemetry.userGrowthType',
+					{ userGrowthType: UserGrowthType.ReturningUser, month: lastMonthDate.getMonth().toString(), year: lastMonthDate.getFullYear().toString()});
+			}
+
+			// if isActiveThisMonth, set wasActiveUserLastMonth
+			if (isActiveThisMonth) {
+				storageService.store('telemetry.wasActiveLastMonth', true, StorageScope.GLOBAL);
+			} else {
+				storageService.store('telemetry.wasActiveLastMonth', false, StorageScope.GLOBAL);
+			}
+
+			// reset the monthly count for the new month
+			storageService.store('telemetry.monthlyUseCount', 0, StorageScope.GLOBAL);
+			storageService.store('telemetry.monthlyLastUseDate', todayString, StorageScope.GLOBAL);
+		} else {
+			// if it's the same month, increment the monthly use count
+			storageService.store('telemetry.monthlyUseCount', monthlyUseCount+1, StorageScope.GLOBAL);
+		}
 	}
 
 	private diffInDays(nowDate: number, lastUseDate: number): number {
 		return (nowDate - lastUseDate) / (24 * 3600 * 1000);
 	}
+
+
+	// Usage Metrics
+	private sendUsageEvent(monthlyUseCount: number, lastMonthDate: Date): void {
+		let userUsageType: UserUsageType;
+		if (monthlyUseCount === 1) {
+			userUsageType = UserUsageType.TireKicker;
+		} else if (monthlyUseCount >= 2 && monthlyUseCount <= 11) {
+			userUsageType = UserUsageType.Occasional;
+		} else if (monthlyUseCount >= 12 && monthlyUseCount <= 20) {
+			userUsageType = UserUsageType.Engaged;
+		} else if (monthlyUseCount > 20) {
+			userUsageType = UserUsageType.Dedicated;
+		}
+		this.telemetryService.publicLog('telemetry.userUsageType',
```

Comment:
I think telemetry.userUsage is enough (especially since we're logging more than just the type)